build.sh

```diff
@@ -28,7 +28,14 @@ find_prunes() {
 }
 
 find_cmd() {
-  echo "find . -type f -and \( -perm +x -or -name '*.sh' \) $(find_prunes)"
+  # GNU `find` dropped compatibility support for the `+` syntax with 4.5.12 (release 2013), it
+  # instead uses `/` which has been supported for a long time. BSD `find` however still used `+`.
+  local perm_format_specifier="/"
+  if [ "$(uname -s)" = "Darwin" ]; then
```

Comment:
I think we should test if the `find ` called is the GNU find or BSD find, because I actually can have the GNU find in my path instead of the BSD's...

Maybe something like `find --version | grep GNU` will work good enough...

What do you think?